Add warning about unsupported HTTP-Redirect binding for SAML 2.0 responses

snowykte0426 opened the pull request:
Spring Security does not support HTTP-Redirect binding for SAML 2.0 responses, as it is not permitted by the SAML specification.

This PR updates the migration guide to explicitly document this limitation using a warning block, helping users avoid potential confusion when integrating with SAML identity providers.

Fixes gh-11161

jzheaux commented:
Hi, @snowykte0426, thanks for this contribution. 

I think this warning should go into a document that is more specific to migrating from the SAML extension to Spring Security. The proposed document is about migrating from Spring Security 6 to 7.

I've added your warning to [the SAML 2.0 Migration Guide](https://github.com/spring-projects/spring-security/wiki/SAML-2.0-Migration-Guide) in the Wiki. It may be valuable to move that Wiki article into the documentation to increase visibility. Would you be able to create a new file in [the SAML 2.0 Documentation](https://github.com/spring-projects/spring-security/tree/main/docs/modules/ROOT/pages/servlet/saml2) and copy the Wiki article text there?

snowykte0426 commented:
Thanks for the feedback, @jzheaux!

I've moved the full content from the SAML 2.0 Migration Guide wiki page into a new AsciiDoc file (`saml2/saml2-migration-guide.adoc`) and added it to the documentation navigation as suggested.

I've also removed the previously added warning from `index.adoc` to avoid duplication.

Let me know if anything else needs to be adjusted!

jzheaux commented on docs/modules/ROOT/pages/servlet/saml2/saml2-migration-guide.adoc:
```diff
@@ -0,0 +1,72 @@
+Use HTTP-POST binding instead when configuring your identity provider.
+====
+
+[[saml2-metadata]]
```
This section isn't particularly relevant anymore, now that Spring Security has more robust metadata support at this point; can we please remove it? I think it would be valuable to add the sample mentioned in this section to the table in the "Examples Matrix" section.

snowykte0426 replied:
Thank you for pointing this out, @jzheaux.

I’ve removed the standalone “Metadata” section and moved its sample link into the “Examples Matrix” table, per your suggestion.  The update is in commit 4ab9cde.

jzheaux commented on docs/modules/ROOT/nav.adoc:
```diff
@@ -89,6 +89,7 @@
 **** xref:servlet/saml2/login/authentication.adoc[SAML2 Authentication Responses]
 *** xref:servlet/saml2/logout.adoc[SAML2 Logout]
 *** xref:servlet/saml2/metadata.adoc[SAML2 Metadata]
+*** xref:saml2/saml2-migration-guide.adoc[SAML 2.0 Migration Guide]
```
Let's please rename this to `saml-extension-migration.adoc` and have the link be "Migrating from Spring Security SAML Extension"

snowykte0426 replied:
Thank you for the guidance, @jzheaux.

I’ve renamed the migration guide file to `saml-extension-migration.adoc` and updated the nav entry to "Migrating from Spring Security SAML Extension" as requested (commit 2388710).

Please let me know if there’s anything else I can adjust!

jzheaux reviewed (changes requested):
Thanks, @snowykte0426! I've left some feedback inline that brings some of the wording up to date.

snowykte0426 commented on docs/modules/ROOT/pages/servlet/saml2/saml2-migration-guide.adoc:
```diff
@@ -0,0 +1,72 @@
+NOTE: This document is a work in progress. Please check back regularly for updates.
```
Thank you for the guidance, @jzheaux.

I’ve removed the work-in-progress note from the SAML Extension migration guide (commit 2fd071c), as it’s implied across our documentation.

snowykte0426 commented on docs/modules/ROOT/pages/servlet/saml2/saml2-migration-guide.adoc:
```diff
@@ -0,0 +1,72 @@
+
+Spring Security SAML Extensions support for Service Providers is provided by a series of filters enabled by adding each filter manually in the correct order to various Spring Security filter chains.
+
+Spring Security’s SAML 2.0 Service Provider support is enabled via the Spring Security `saml2Login` and `saml2Logout` DSL methods. It selects the correct filters to add and puts them in the appropriate places in the filter chain.
```
Thank you for the suggestion, @jzheaux.

I’ve updated the “Login & Logout” section so that each DSL method now links to its reference page:
- `saml2Login` → xref:servlet/saml2/login/index.adoc[`saml2Login`]
- `saml2Logout` → xref:servlet/saml2/logout.adoc[`saml2Logout`]
- `saml2Metadata` → xref:servlet/saml2/metadata.adoc[`saml2Metadata`]

This change is in commit 862f7c4.

snowykte0426 commented on docs/modules/ROOT/pages/servlet/saml2/saml2-migration-guide.adoc:
```diff
@@ -0,0 +1,72 @@
+NOTE: This document is a work in progress. Please check back regularly for updates.
+
+This document contains guidance for moving SAML 2.0 Service Providers from Spring Security SAML Extensions 1.x to Spring Security 5.7.x. Since Spring Security doesn’t provide Identity Provider support, migrating a Spring Security SAML Extensions Identity Provider is out of scope for this document.
```
Thank you for the suggestion, @jzheaux.

I’ve removed the version suffix “5.7.x” from all references to Spring Security to simplify maintenance (commit b60b06d)

jzheaux commented:
Thank you for the updates, @snowykte0426! In preparation for merging, will you please squash your commits? 

Since you are on `main`, you might want to instead create a branch, push your squashed commit to that branch, and create a new PR based on that branch. In this way, you won't have to do a force push on your `main` branch.

snowykte0426 commented:
Closing in favor of #17076 after squashing commits into a single one per feedback.
